Add the Handle Survey Results help topics (#4631)
surveyjs/survey-library: docs/handle-survey-results-store.md

# Store Survey Results

Survey results are JSON objects that you can store in your own database or in the SurveyJS Service.

- [Store Survey Results in Your Own Database](#store-survey-results-in-your-own-database)
- [Store Survey Results in the SurveyJS Service](#store-survey-results-in-the-surveyjs-service)

## Store Survey Results in Your Own Database

> To store survey results in your own database, handle the [onComplete](https://surveyjs.io/Documentation/Library?id=surveymodel#onComplete) event. It is raised when a user clicks the Complete button. The following code shows how to use this event to send survey results to your server:

Reviewer: ```suggestion
To store survey results in your own database, handle the [onComplete](https://surveyjs.io/Documentation/Library?id=surveymodel#onComplete) event raised in response to a click on the Complete button. The following code shows how to use this event to send survey results to your server:
```

```js
import { Model } from "survey-core";

const surveyJson = { ... };

const survey = new Model(surveyJson);
survey.onComplete.add(function (sender, options) {
  // Display the "Saving..." message (pass a string value to display a custom message)
  options.showDataSaving();
  const xhr = new XMLHttpRequest();
  xhr.open("POST", "your/server/url");
  xhr.setRequestHeader("Content-Type", "application/json; charset=utf-8");
  xhr.onload = xhr.onerror = function () {
    if (xhr.status == 200) {
      // Display the "Success" message (pass a string value to display a custom message)
      options.showDataSavingSuccess();
      // Alternatively, you can clear all messages:
      // options.showDataSavingClear();
    } else {
      // Display the "Error" message (pass a string value to display a custom message)
      options.showDataSavingError();
    }
  };
  xhr.send(JSON.stringify(sender.data));
});
```

> The `onComplete` event handler only sends survey results in JSON format to your server. How you store them fully depends on your backend.

Reviewer: ```suggestion
The `onComplete` event handler only sends survey results in JSON format to your server. The way you store them fully depends on your backend.
```
